### PR TITLE
feat: webhook history view

### PR DESCRIPTION
**Pull request by thisisnithin**
## Motivation and Context

https://github.com/user-attachments/assets/e3ea9876-d720-4ae9-8402-94fdc6dd8dfe

Update
image

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: controlplane/src/core/bufservices/PlatformService.ts [hunk 1]
```diff
@@ -11449,5 +11499,122 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
...
+
+      return handleError<PlainMessage<RedeliverWebhookResponse>>(ctx, logger, async () => {
+        const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
+        logger = enrichLogger(ctx, logger, authContext);
+
+        const orgRepo = new OrganizationRepository(logger, opts.db, opts.billingDefaultPlanId);
+        const redeliverWebhookService = new RedeliverWebhookService(opts.db, authContext.organizationId, logger);
+
+        const originalDelivery = await orgRepo.getWebhookDeliveryById(req.id, authContext.organizationId);
+        if (!originalDelivery) {
```
Comment (StarpTech) [inline controlplane/src/core/bufservices/PlatformService.ts:11601]: Only users with write / dev access should be able to do it.

File: controlplane/src/core/bufservices/PlatformService.ts [hunk 2]
```diff
@@ -11449,5 +11499,122 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
...
+          response: {
+            code: EnumStatusCode.OK,
+          },
+          deliveries,
+          totalCount,
+        };
+      });
+    },
+
+    getDeliveryDetails: (req, ctx) => {
```
Comment (StarpTech) [inline controlplane/src/core/bufservices/PlatformService.ts:11551]: Get WebhookDeliveryDetails?

Comment (thisisnithin) [reply controlplane/src/core/bufservices/PlatformService.ts]: Done

File: controlplane/src/core/bufservices/PlatformService.ts (hunk 1 above)
Comment (thisisnithin) [reply controlplane/src/core/bufservices/PlatformService.ts:11610]: Done

File: controlplane/src/core/webhooks/OrganizationWebhookService.ts [hunk 3]
```diff
@@ -344,29 +344,72 @@ export class OrganizationWebhookService {
...
+        type: config.type,
+        endpoint: config.url,
+        eventName,
+        payload: JSON.stringify(data),
+        createdById: actorId,
+        requestHeaders: {},
+      };
+
+      axiosRetry(this.httpClient, {
+        retries: 5,
```
[the remark this answers is not in the document]
Comment (thisisnithin) [reply controlplane/src/core/webhooks/OrganizationWebhookService.ts:363]: Do we want to make it 6 or do we have 6 for something else?

File: controlplane/src/core/webhooks/utils.ts [hunk 4]
```diff
@@ -17,10 +17,12 @@ export const makeWebhookRequest = async <Data = any>(
     headers['X-Cosmo-Signature-256'] = createHmac('sha256', signatureKey).update(dataString).digest('hex');
   }
 
-  await axiosInstance.post(url, data, {
+  const res = await axiosInstance.post(url, data, {
     headers,
     timeout: 10_000,
```
[the remark this answers is not in the document]
Comment (thisisnithin) [reply controlplane/src/core/webhooks/utils.ts]: Done

File: controlplane/src/core/webhooks/OrganizationWebhookService.ts [hunk 5]
```diff
@@ -70,13 +71,6 @@ export class OrganizationWebhookService {
     this.httpClient = axios.create({
       timeout: 10_000,
```
[the remark this answers is not in the document]
Comment (thisisnithin) [reply controlplane/src/core/webhooks/OrganizationWebhookService.ts]: Done

File: controlplane/src/core/webhooks/RedeliverWebhookService.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        shouldResetTimeout: true,
+        onRetry: (count) => {
+          retryCount = count;
+        },
+      });
+
+      try {
+        const res = await this.httpClient.post(originalDelivery.endpoint, JSON.parse(originalDelivery.payload), {
+          headers: originalDelivery.requestHeaders,
+          timeout: 10_000,
```
[the remark this answers is not in the document]
Comment (thisisnithin) [reply controlplane/src/core/webhooks/RedeliverWebhookService.ts]: Done

File: controlplane/src/core/webhooks/utils.ts [hunk 7]
```diff
@@ -17,10 +17,12 @@ export const makeWebhookRequest = async <Data = any>(
...
   });
+
+  return res;
```
[the remark this answers is not in the document]
Comment (thisisnithin) [reply controlplane/src/core/webhooks/utils.ts]: Done

File: controlplane/src/db/schema.ts [hunk 8]
```diff
@@ -1127,6 +1127,48 @@ export const webhookGraphSchemaUpdate = pgTable(
...
+  organizationId: uuid('organization_id')
+    .notNull()
+    .references(() => organizations.id, {
+      onDelete: 'cascade',
+    }),
+  type: webhookDeliveryType('type').notNull(),
+  endpoint: text('endpoint').notNull(),
+  eventName: text('event_name').notNull(),
+  payload: text('payload').notNull(),
+  requestHeaders: customJson<Record<string, any>>('request_headers').notNull(),
```
[the remark this answers is not in the document]
Comment (thisisnithin) [reply controlplane/src/db/schema.ts]: Improved

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**